spec/javascript/controllers/analytics_spec_helper.js

```diff
@@ -15,53 +15,135 @@ export default class {
-
     describe("with cookie already set", () => {
-      beforeEach(() => { this.setAcceptedCookie() })
+      beforeEach(() => {
+        delete window.willRedirectionOccurByVWO ;
+        Cookies.remove(CookiePreferences.cookieName) ;
+      })
+
+      describe("When VWO has completed", () => {
+        beforeEach(() => {
+          window.willRedirectionOccurByVWO = false ;
+          this.setAcceptedCookie()
+        })
+
+        describe("with no service id", () => {
+          beforeEach(() => { this.initApp(name, controller, "") })
 
-      describe("with no service id", () => {
-        beforeEach(() => { this.initApp(name, controller, "") })
+          it("Should not register the service", () => {
+            expect(typeof(window[serviceFunctionName])).toBe("undefined") ;
+          }) ;
+        })
+
+        describe("with service id set", () => {
+          beforeEach(() => { this.initApp(name, controller, "1234") }) ;
+
+          it("Should register the service", () => {
+            expect(typeof(window[serviceFunctionName])).toBe("function") ;
+          }) ;
+        })
+      })
+
+      describe("When VWO has completed but is navigating away", () => {
+        beforeEach(() => {
+          window.willRedirectionOccurByVWO = true ;
+          this.initApp(name, controller, "1234")
+        }) ;
 
         it("Should not register the service", () => {
           expect(typeof(window[serviceFunctionName])).toBe("undefined") ;
-        }) ;
-      })
+        })
+      }) ;
 
-      describe("with service id set", () => {
-        beforeEach(() => { this.initApp(name, controller, "1234") }) ;
+      describe("VWO has not yet completed", () => {
+        beforeEach(() => {
+          this.setAcceptedCookie() ;
+          this.initApp(name, controller, "1234")
+        })
 
-        it("Should register the service", () => {
-          expect(typeof(window[serviceFunctionName])).toBe("function") ;
+        describe("with service id set", () => {
+          it("Should not register the service", () => {
+            expect(typeof(window[serviceFunctionName])).toBe("undefined")
+          })
         }) ;
-      })
+
+        describe("Than VWO completes", () => {
+          it("Should now register the service", () => {
+            this.vwoCompletes() ;
+            expect(typeof(window[serviceFunctionName])).toBe("function")
+          })
+        })
+
+        describe("Than VWO completes but is navigating away", () => {
+          it("Should now register the service", () => {
+            this.vwoCompletesAndRedirects() ;
+            expect(typeof(window[serviceFunctionName])).toBe("undefined")
+          })
+        })
+      }) ;
     })
   }
 
   static describeWhenEventFires(name, controller, serviceFunctionName, cookieCategory) {
-    beforeEach(() => { Cookies.remove(CookiePreferences.cookieName) })
-
-    describe("without cookie set yet", () => {
+    describe("when cookie:accepted event first", () => {
```

Comment:
I had to do a double take on this, I think it would be clearer as:

> when cookies have not yet been accepted

Also found specs on line 131 and 139 a little hard to follow, could we do anything to make the intention more explicit in the spec names, for example instead of:

> VWO then completes and emits event

something like:

> registers the service only after cookies are accepted and VWO completes

and:

> VWO then completes but is navigating away

could be:

> does not register the service if cookies are accepted and VWO redirects

Reply:
Good call - this was me worrying more about figuring out the sequence last night and not paying enough attention to naming